.github/workflows/edge_tests.yml

```diff
@@ -93,4 +95,4 @@ jobs:
         sudo /sbin/sysctl -w net.ipv4.ping_group_range="0 2147483647"
     - name: Test
       run: |
-        make test_${{ matrix.target }}
+        make init_proof test_${{ matrix.target }}
```

Comment:
This relies on `init_proof` being executed before `test_...`. However, while true in this instance,  `make` is not obliged to do it in that order (especially not when parallelism is involved). It would be better to make $(proof_archive) a dependency of the test targets and let `make` figure out the rest.

Reply:
> This relies on init_proof being executed before test_.... However, while true in this instance, make is not obliged to do it in that order (especially not when parallelism is involved).

Does this mean, in general for a call of `make a b` (with `a` and `b` being phony targets), `make` could execute `b` before `a`?

> It would be better to make $(proof_archive) a dependency of the test targets and let make figure out the rest.

That doesn't make sense to me. `$(proof_archive)` is the `proof.tar.zst`. Making it a dependency of the test targets would recreate this archive before running the tests. What `init_proof` does is extracting this archive, so that the proof sessions can be used in the tests. This is an optional step. You don't want to always overwrite the proof sessions before executing tests. You also don't want to always use existing proof sessions, e.g., when the intention is to run GNATprove from a clean state.

I suppose you would like to have some `$(extracted_proof_session_file)` as dependency for the test targets, so that the archive is extracted before the tests, if this file is missing. `clean_proof` could then not just remove all proof sessions, but would also have to create an "empty archive", which can be extracted before the next tests. Would that make sense? The disadvantage of this approach would be that the tests could not be used without Git LFS.

Reply:
I have changed the behavior as discussed yesterday (5161a7b385884b943989bda17d5658dccd3dc8df) and refactored the Makefiles to remove duplicate rules (7bfc2a14ee24ec2adbaa4c85302d4a42532dd7a1).